mlflow/store/file_store.py

```diff
@@ -21,10 +21,11 @@
 from mlflow.utils.file_utils import (is_directory, list_subdirs, mkdir, exists, write_yaml,
                                      read_yaml, find, read_file_lines, read_file, build_path,
                                      write_to, append_to, make_containing_dirs, mv, get_parent_dir,
-                                     list_all)
+                                     list_all, parse_path)
 from mlflow.utils.mlflow_tags import MLFLOW_RUN_NAME, MLFLOW_PARENT_RUN_ID
 
 _TRACKING_DIR_ENV_VAR = "MLFLOW_TRACKING_DIR"
+_LOCAL_FS_URI_PREFIX = "file:///" if os.sep == "/" else "file://"
```

Comment:
this was removed, apologies I think there was a bad rebase on my part. It is supposed to be file:///

Reply:
ah ok, no worries :)

Reply:
Actually, it was leftover but also is related to the local path vs absolute path issue mentioned before about file:/// forcing absolute paths.

I am starting to think parse.urlparse is the problem in my current implementation. If we add the file:// prefix. And require it for local artifact repo and file store(supporting users specifying path for linux but returning the uri with the correct prefix), then we can parse out file://, and handle the path accordingly. Windows paths will be supported, linux paths will also be. 

The uri would be incorrect for the relative path scenario, but distinguishing between relative and absolute paths does not make sense otherwise unless we allow file://// for absolute paths, which seems off to me

Reply:
I think file:///... or file://localhost/... are meant to be absolute paths. Relative paths would omit the file:/// prefix altogether.

E.g. check the answers in https://stackoverflow.com/questions/7857416/file-uri-scheme-and-relative-files.

I think we should always standardize the uri to be absolute path with file:/// prefix. What do you think?
I am not sure if windows path would need to have slashes converted in that case? I think they might.